core/src/main/scala/dagr/core/execsystem/TaskManager.scala

```diff
@@ -559,8 +563,8 @@ class TaskManager(taskManagerResources: SystemResources = TaskManagerDefaults.de
     val runningTasks: Map[UnitTask, ResourceSet] = runningTasksMap
 
     // get the tasks that are eligible for execution (tasks with no dependents)
-    val (emptyTasks: List[Task], readyTasks: List[Task]) = {
-      graphNodesInStateFor(NO_PREDECESSORS).map(_.task).toList.partition(_.isInstanceOf[Task.EmptyTask])
+    val (emptyTasks: Seq[Task], readyTasks: Seq[Task]) = {
+      graphNodesInStateFor(NO_PREDECESSORS).map(_.task).toSeq.partition(_.isInstanceOf[Task.EmptyTask])
```

Comment:
As a side note: I really think we can speed up dagr further by maintaining a bit more internal state.  `graphNodesInStateFor` loops through all the nodes, versus just having a simple lookup.  This code really shows how poor my first attempt at Scala was.

Reply:
Yikes.  Maybe log an issue to come back and fix this some day?